resources/speech-detector/src/test/java/org/restcomm/media/resource/speechdetector/NoiseThresholdDetectorTest.java

```diff
@@ -1,61 +1,61 @@
- * Lesser General Public License for more details.
- *
- * You should have received a copy of the GNU Lesser General Public
- * License along with this software; if not, write to the Free
- * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
- * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
- */
-
-package org.restcomm.media.resource.speechdetector;
-
-import static junit.framework.TestCase.assertTrue;
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertFalse;
-
-import org.junit.After;
-import org.junit.Before;
-import org.junit.Test;
-
-/**
-  * @author Vladimir Morosev (vladimir.morosev@telestax.com)
-  */
-public class SpeechDetectorTest {
-
-    private SpeechDetector testee;
-
-    @Before
-    public void setUp() {
-        testee = new SpeechDetector(10);
-    }
-
-    @After
-    public void cleanUp() {
-        testee = null;
-    }
-
-    @Test
-    public void testSilence() {
-    	final byte[] bytes = new byte[] { 0x00, 0x00, 0x00, 0x00 };
-        assertTrue(testee.detect(bytes, 0, bytes.length));
-    }
-
-    @Test
-    public void testSpeech() {
-    	final byte[] bytes = new byte[] { 0x00, 0x0f, 0x70, 0x7f };
-        assertFalse(testee.detect(bytes, 0, bytes.length));
-    }
-
-}
+/*
+ * TeleStax, Open Source Cloud Communications
+ * Copyright 2011-2017, Telestax Inc and individual contributors
+ * by the @authors tag.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.restcomm.media.resource.speechdetector;
+
+import static junit.framework.TestCase.assertTrue;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+  * @author Vladimir Morosev (vladimir.morosev@telestax.com)
+  */
+public class NoiseThresholdDetectorTest {
+
+    private SpeechDetector testee;
+
+    @Before
+    public void setUp() {
+        testee = new NoiseThresholdDetector(10);
+    }
+
+    @After
+    public void cleanUp() {
+        testee = null;
+    }
+
+    @Test
+    public void testSilence() {
+    	final byte[] bytes = new byte[] { 0x00, 0x00, 0x00, 0x00 };
+        assertTrue(testee.detect(bytes, 0, bytes.length));

```

Comment:
you're testing for silence, but detect method returns true? shouldn't return false?

Reply:
wouldn't be more meaningful to use a byte array with some payload that is considered under the silenceLevel (10). Because the threshold is the determining factor in this algorithm.

Reply:
write test in a given-when-then format

Reply:
I will make similar approach as it is done for Opus codec. Prerecorded PCM coded files will be used there as a test resource.

Reply:
I guess a single packet is necessary, as this algorithm uses 20ms frames of audio. Let's keep it simple and wrap this one up.